### PR TITLE
Feat: IPFS pinata pin

### LINKED ISSUE #251: IPFS Pinata Pin
**User Story**
As a proposer, I dont want to rely on local ipfs instance pinning to replicate and retrieve data. 

**Alternative Solutions**
If you look at dxvote we have pinata pinning services along with using the local ipfs node. We want that in DAVI too, we should use two endpoints listed below. We can also reuse the key in the file linked below. 

We should also remove the duplicated list of gateways we have currently. 

Endpoints:
- https://api.pinata.cloud/pinning/pinByHash
- https://api.pinata.cloud/pinning/pinJSONToIPFS

Example: https://github.com/DXgovernance/dxvote/blob/6f5edb30b6e727c5574750fa6831f3b9940a7656/src/services/PinataService.ts

New gateway: https://davi.mypinata.cloud/ipfs/

### PR DESCRIPTION
**Pull request by dcrescimbeni**
# Description

Proposal data now gets pinned to Pinata service.

If for some reason the content can't be pinned, it shows a modal, prompting to retry the upload, make the proposal without uploading the file, or close the modal to edit the proposal.

To trigger this behavior, modify the API key of the `usePinataIPFS` hook to something invalid. If the content of the proposal can't be uploaded to IPFS, the description won't load and it will show an error message instead.

Added two new IPFS API endpoints to retrieve files:
`https://davi.mypinata.cloud/`
`https://ipfs.fleek.co/`

Some minor changes:
1. Deleted IPFS util function since it was unused. Also removed `axios` library since that was the only place it was used.
2. Moved status circles (like the ones showed in the Tenderly simulation) to their own folder in `primitives`
3. Added translations

Closes #251 

https://www.loom.com/share/b1595fa6fcb34cb1a069c1a86d563473

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by dcrescimbeni** [comment]
Synpress tests are passing locally

**Comment by Filipv95** [comment]
Hey, @dcrescimbeni how can this be tested?

**Comment by dcrescimbeni** [comment]
> Hey, @dcrescimbeni how can this be tested?

We can test this using the DevTools:

1. Open the DevTools
2. Create a new proposal
3. Go to the "Network" tab
4. Look for "pinJSONToIPFS" network transaction
5. Click on it, and go to the "Payload" section. The "pinataContent" property should have the title and options of the proposal.
6. Inside the property "pinataMetadata" > "name", after "DescriptionHash", there's a string representing the content hash of that proposal, looks like "QmUQBdVCUheoByNtwSBV3YALsqX7BbvPeJ9gyKjpzHnKvU"
7. Copy that hash
8. Go to "https://davi.mypinata.cloud/ipfs/QmUQBdVCUheoByNtwSBV3YALsqX7BbvPeJ9gyKjpzHnKvU" (replace with the new proposal hash)
9. You should see the proposal title and options

**Comment by Filipv95** [comment]
Hey @dcrescimbeni tnx for the explanation!

I've was not able to find "pinJSONToIPFS" in network transactions, but used filter 'ipfs' instead (within Network tab) and got only one request, which contained hash of my newly created proposal. 
When I copied it to link you provided I was able to see Proposal title and options.
What am I doing wrong, why can't I see transaction you've mentioned in Network tab? 
 
ipfs

**Comment by dcrescimbeni** [comment]
@Filipv95 , I think I know where the confusion is

The `pinJSONToIPFS` information in the network tab gets shown when creating a new proposal, since that's the time that the content gets uploaded to Pinata.

I made a video that shows the steps.

If you still can't replicate it, let me know here or ping me on keybase.

https://www.loom.com/share/4628d16da727495f88d56f65bd055dd2

**Comment by Filipv95** [comment]
@dcrescimbeni Tnx, I've managed to replicate, on DXD Guild now. Earlier I was creating Proposal on SWPR Guild and there I had no this (`pinJSONToIPFS`) transaction under Network tab, is this expected behavior?

**Comment by dcrescimbeni** [comment]
> @dcrescimbeni Tnx, I've managed to replicate, on DXD Guild now. Earlier I was creating Proposal on SWPR Guild and there I had no this (`pinJSONToIPFS`) transaction under Network tab, is this expected behavior?

No, it should be working irrespective of the guild.

I can see the pinJSONToIPFS while creating a proposal in SWAPR guild

image

File: src/hooks/Guilds/ipfs/usePinataIPFS.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { useTypedParams } from 'Modules/Guilds/Hooks/useTypedParams';
+import { useTranslation } from 'react-i18next';
+
+const DEFAULT_API_KEY =
```
Comment (mprasanjith) [inline src/hooks/Guilds/ipfs/usePinataIPFS.ts:4]: Are we hardcoding this?

**Review by mprasanjith — APPROVED** [review]
Looks great! Not sure if we want to hardcode the API key here.

File: src/hooks/Guilds/ipfs/usePinataIPFS.ts (hunk 1 above)
Comment (dcrescimbeni) [reply src/hooks/Guilds/ipfs/usePinataIPFS.ts:4]: It was hardcoded in dxvote, that's why I put it here.

But I guess at the end of the day we're sending the API key in out post requests, so I'm not sure how effective it'll be to put it in a `.env`

**Comment by Filipv95** [comment]
@dcrescimbeni You are right! I've found it now, not sure why I couldn't yesterday, anyway good job!

**Pull request merged**